### PR TITLE
Delete author

### LINKED ISSUE #6: User Story 20, User can delete an author
```
As a Visitor,
When I visit an author's show page,
I see a link on the page to delete the author.
This link should return me to the book index page where I
no longer see this author listed.
If this author was the only author for any book, that book is also deleted.
If this author co-authored a book with someone else, that book should also be deleted, but not the other author.

(you may need to delete other content before you can delete an author or book)
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written
- [ ] edge case code written
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #9

child of #33

depends on #14

### PR DESCRIPTION
**Pull request by juliamarco**
Adds test and functionality for deleting author.

@iandouglas Had a lot of trouble with getting the dependent destroyed. It's working now as per line 9 in authors_controller. We have a feeling this could be done in a better way. Thanks for letting us know any suggestions.

closes #6

### CODE REVIEW

File: spec/features/authors/authors_show_spec.rb [hunk 1]
```diff
@@ -42,5 +42,17 @@
...
+      visit author_path(@herman_melville)
+
+      click_button("Delete Author")
+
+      expect(current_path).to eq(books_path)
+
+      expect(page).to_not have_css("#book-#{@book_1.id}")
+      expect(page).to have_content(@stephen_king.name)
+      expect(page).to have_content(@book_2.title)
+    end
```
Comment (iandouglas) [inline spec/features/authors/authors_show_spec.rb]: should also expect page to not have content of @herman_melville.name as well
